spec/models/document_spec.rb

```diff
@@ -217,25 +217,34 @@
   end
 
   describe "after_create" do
+    include ActiveJob::TestHelper
+
     context "when the file extension is .heic" do
-      it "creates a job to convert the file to jpg" do
-        document = build :document, upload_path: Rails.root.join("spec", "fixtures", "files", "IMG_4851.HEIC")
-        allow(HeicToJpgJob).to receive(:perform_later)
+      before do
+        allow(HeicToJpgJob).to receive(:perform_later).and_call_original
+      end
 
+      it "creates a job to convert the file to jpg and delays ActiveStorage::AnalyzeJob until then" do
+        document = build :document, upload_path: Rails.root.join("spec", "fixtures", "files", "IMG_4851.HEIC")
         document.save!
 
+        expect(ActiveJob::Base.queue_adapter.enqueued_jobs.map { |x| x["job_class"] }).not_to include("ActiveStorage::AnalyzeJob")
```

Comment:
could we instead assert here that the `enqueued_jobs` is just a `[HeicToJpgJob]`

Reply:
done, yay